internal/controller/storageclient_controller.go

```diff
@@ -609,6 +779,64 @@ func (r *storageClientReconcile) reconcileClientStatusReporterJob() error {
 	return nil
 }
 
+func (r *storageClientReconcile) hasPersistentVolumes(clientProfileNames []string) (bool, error) {
```

Comment:
Cant we implement a single func that handles all of the `has` cases using `PartialMetadataObjectList`?

Reply:
not possible as we have index on the volumehand keyed by the clientprofile and so we can't do w/ partial obj.

Reply:
deferred as agreed but need to optimize this as there could be a lot of PVs in the cluster and all we need is to only know the presence rather than ballooning the cache.